jenkins_pipelines/scripts/maintenance_json_generator.py

```diff
@@ -148,174 +152,160 @@
 # Dictionary for SUMA 5.0 Server and Proxy, which is then added together with
 # the common dictionary for 4.3 client tools and 5.0 BETA client tools. Both
 # client tools are needed until 5.0 gets its own client tools.
-nodesdict50 = {
-    "server": ["/SUSE_Products_SUSE-Manager-Server_5.0_x86_64/",
-               "/SUSE_Updates_SUSE-Manager-Server_5.0_x86_64/"],
-    "proxy": ["/SUSE_Products_SUSE-Manager-Proxy_5.0_x86_64/",
-              "/SUSE_Updates_SUSE-Manager-Proxy_5.0_x86_64/"]
+nodesdict50: dict[str, set[str]] = {
+    "server": {"/SUSE_Products_SUSE-Manager-Server_5.0_x86_64/",
+               "/SUSE_Updates_SUSE-Manager-Server_5.0_x86_64/"},
+    "proxy": {"/SUSE_Products_SUSE-Manager-Proxy_5.0_x86_64/",
+              "/SUSE_Updates_SUSE-Manager-Proxy_5.0_x86_64/"}
 }
 
 # Merging clienttools43dict and clienttools50betadict
 # For now we need non-BETA together with BETA client tools until the split
 # between 4.3 and 5.0 for client tools happens, just before GA
 # After the split the 5.0 non-BETA client tools will have different names than
 # the 4.3 non-BETA client tools. They will not be common.
-finaldict = {}
+finaldict: dict[str, set[str]] = {}
 
 for key, value in clienttools43dict.items():
+    finaldict[key] = value
     if key in clienttools50betadict:
-        if isinstance(value, list):
-            if isinstance(clienttools50betadict[key], list):
-                finaldict[key] = value + clienttools50betadict[key]
-            else:
-                finaldict[key] = value + [clienttools50betadict[key]]
-        else:
-            if isinstance(clienttools50betadict[key], list):
-                finaldict[key] = [value] + clienttools50betadict[key]
-            else:
-                finaldict[key] = [value, clienttools50betadict[key]]
-    else:
-        finaldict[key] = value
-
+        finaldict[key].update(clienttools50betadict[key])
```

Comment:
You can consider using something more pythonic like:
```
finaldict: dict[str, set[str]] = {key: value.union(clienttools50betadict.get(key, set())) for key, value in clienttools43dict.items()}
```
similarly below.

Reply:
I'm fine replacing it with a comprehension but others may find it more readable as its is.
I'll check later for comments on this.